modules/dhcp_common/record.rb

```diff
@@ -34,25 +33,19 @@ def [] opt
 
     #TODO move this away from here, as this suppose to be a generic interface
     def deleteable?
-      @deleteable
+      !!@options[:deleteable]
     end
 
     def <=>(other)
       self.ip <=> other.ip
     end
 
-    def ==(other)
-      options == other.options
+    def eql?(other)
+      self == other
     end
 
-    # compare between incoming request and our existing record
-    # if our record has all requested attributes then we say they are comparable
-    def self.compare_options(record, request)
-      request.each do |k,v|
-        return false if record[k.to_sym] != v
-      end
-      true
+    def ==(other)
+      ip == other.ip && mac == other.mac && subnet == other.subnet && deleteable? == other.deleteable? && options == other.options
```

Comment:
Should we check if other is really a record before comparison? Same for `Proxy::DHCP::Lease`.